Add CONTRIBUTION file (#100)
OpenLiberty/guide-maven-intro: CONTRIBUTING.md

# Contributing to Open Liberty Guides

Anyone can contribute to the Open Liberty guides and we welcome your contributions!

There are multiple ways to contribute: report bugs, fix bugs, contribute code, improve upon documentation, etc. 

## Raising Issues

Please raise any bug reports on the Open Liberty guide's repository: https://github.com/OpenLiberty/[guide-name]/issues, where `[guide-name]` is the repository name of the guide . For new topics, large updates to existing guides, or general suggestions and ideas, report your issue on the [Open Liberty common guides repository](https://github.com/OpenLiberty/guides-common/issues). Be sure to search the list to see if your issue has already been raised.

A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong. Provide as much context as possible so we can try to recreate the issue.

## Contributions 

> Contributing to the Open Liberty guides is simple. All you need to do is create your own branch to include your contribution(s). If you want to make updates to existing guides, create a new branch and open a pull request to the `qa` (quality assurance) branch.

Reviewer: - [ ] "...to existing guides, create a new branch..." it should not be possible to have a branch to update guides.

Author: Isn't the contributor creating a new branch for their changes?

Reviewer: will be addressed by gkwan-ibm-patch-1 branch

### Text or Content Contributions

If you are contributing text of content changes via a pull request and you certify the [Developer Certificate of Origin (DCO)](https://developercertificate.org)), add a line to the end of the Git commit message to sign your work:

```
Signed-off-by: Jane Williams <jane.williams@gmail.com>
```

The sign-off is just a line at the end of the commit message that certifies that you wrote it or otherwise have the right to pass it on as an open source patch.

> Use your real name when you sign. We can't accept pseudonyms or anonymous contributions. To open a pull request to the `qa` branch, follow the official [git documentation](https://git-scm.com/docs/git-request-pull). If reviewers agree with your change, your change will be merged to the `master` branch for publishing.

Reviewer: - [ ] I don't think need the git documentation and I doubt the example has ` v1.0` will work for us

Author: This was from when you suggested to add instructions on cloning, creating a PR, etc... I removed nonetheless.

### Code Contributions

> If you are contributing trivial changes such as typos, redundant spaces, minor formatting and spelling errors, you must follow the above instructions under [Text or Content Contributions](https://github.com/OpenLiberty/guide-maven-intro/blob/dco-file/CONTRIBUTING.md#text-or-content-contribution).

Reviewer: - [ ] the link not work

Reviewer: will be addressed by gkwan-ibm-patch-1 branch

If you are contributing code changes via a pull request for anything except trivial changes, you must signoff on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf) If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf). Instructions how to sign and submit these agreements are located at the top of each document. 

After we obtain the signed CLA, you are welcome to open a pull request against the `qa` branch, and the team will be notified for review. We ask you follow these steps through the submission process.

1. Open PR's against the `qa` branch, as we ensure changes pass our series of verification buckets before pushing to master.
2. A "CLA signed" label will be added.

> 3. A team of "reviewers" will be notified and will review your change. If approved, a full integration build will be invoked.

Reviewer: - [ ] I don't think we do "...a full integration build will be invoked." It is for OL build.

Reviewer: will be addressed by gkwan-ibm-patch-1 branch

If reviewers agree with your change, your change will be merged to the `master` branch for publishing.

### Code Contributions

If you are contributing trivial changes such as typos, redundant spaces, minor formatting and spelling errors, you must follow the above instructions under [Text or Content Contributions](./CONTRIBUTING.md#text-or-content-contributions).

If you are contributing code changes via a pull request for anything except trivial changes, you must signoff on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf) If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf). Instructions how to sign and submit these agreements are located at the top of each document. 

After we obtain the signed CLA, you are welcome to open a pull request against the `qa` branch, and the team will be notified for review. We ask you follow these steps through the submission process.

1. Open a pull request against the `qa` branch.

> 2. A "CLA signed" label will be added.

Reviewer 2: How? Does it happen automatically or do we manually do it when we receive the PR? We should say so that they know what to expect and whether they need to wait for it to happen.

Author: Addressed:
```
2. A "CLA signed" label will be manually added by the team.
```

3. The team will be notified and will review your change. 
    - If there are further changes to be made, the team will request changes on the pull request. 
    - If the team does not agree with the change, the PR will be closed with an explanation and suggestion for follow-up.

> 4. If the team approves, a full Open Liberty site build will be invoked.

Reviewer 2: "invoked" -> "run" or "started"? (just a bit clearer)

Author: Addressed.

> 5. Based on the results of the build, and if further review is needed, more discussion will occur.

Reviewer 2: "more discussion will occur" -> "we will discuss it with you" or something like that - be more specific about who will do what so it's clearer what's happening if things go quiet for a bit.

Author: Addressed:
```
5. Based on the results of the build:
    - If further review is needed, we will let you know about a pending review from our team and discuss any necessary improvements that need to be made to your change(s). 
    - If everything is successful, the team will merge your PR.
```

# Contributing to Open Liberty Guides

Anyone can contribute to the Open Liberty guides and we welcome your contributions!

There are multiple ways to contribute: report bugs, fix bugs, contribute code, improve upon documentation, etc. 

## Raising Issues

Please raise any bug reports on this [guide repository](../../issues). For new topics, large updates to existing guides, or general suggestions and ideas, report your issue on the [Open Liberty common guides repository](https://github.com/OpenLiberty/guides-common/issues). Be sure to search the list to see if your issue has already been raised.

A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong. Provide as much context as possible so we can try to recreate the issue.

## Contributions 

Contributing to an Open Liberty guide is simple. All you need to do is create your own branch to include your contribution and open a pull request to the `qa` (quality assurance) branch.

### Text or Content Contributions

> If you are contributing text of content changes via a pull request and you certify the [Developer Certificate of Origin (DCO)](https://developercertificate.org), add a line to the end of the Git commit message to sign your work:

Reviewer 3: ```suggestion
If you are contributing text of content changes via a pull request, all you need is to certify the originality of the work via a [Developer Certificate of Origin (DCO)](https://developercertificate.org).

Add a line to the end of the Git commit message to sign your work:
```

```
Signed-off-by: Jane Williams <jane.williams@gmail.com>
```

The sign-off is just a line at the end of the commit message that certifies that you wrote it or otherwise have the right to pass it on as an open source patch.

Use your real name when you sign. We can't accept pseudonyms or anonymous contributions. 

Many Git UI tools have support for adding the `Signed-off-by` line to the end of your commit message. This line can be automatically added by the `git commit` command by using the `-s` option.

If reviewers agree with your change, your change will be merged to the `master` branch for publishing.

### Code Contributions

> If you are contributing trivial changes such as typos, redundant spaces, minor formatting and spelling errors, you must follow the above instructions under [Text or Content Contributions](./CONTRIBUTING.md#text-or-content-contributions).

Reviewer 3: ```suggestion
If you are contributing trivial code changes such as typos, redundant spaces, minor formatting and spelling errors, follow the same instructions under [Text or Content Contributions](./CONTRIBUTING.md#text-or-content-contributions).
```

> If you are contributing code changes via a pull request for anything except trivial changes, you must signoff on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf) If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf). Instructions how to sign and submit these agreements are located at the top of each document. 

Reviewer 3: ```suggestion
If you are contributing code changes via a pull request for non-trivial changes, you must sign off on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf).

If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf).

Instructions how to sign and submit these agreements are located at the top of each document. 
```

Reviewer 3: Can we provide examples of non-trivial changes too?

After we obtain the signed CLA, you are welcome to open a pull request against the `qa` branch, and the team will be notified for review. We ask you follow these steps through the submission process.

1. Open a pull request against the `qa` branch.
2. A "CLA signed" label will be manually added by the team.
3. The team will be notified and will review your change(s). 
    - If there are further changes to be made, the team will request changes on the pull request. 
    - If the team does not agree with the change, the PR will be closed with an explanation and suggestion for follow-up.
4. If the team approves, a full Open Liberty site build will be run.
5. Based on the results of the build:
    - If further review is needed, we will let you know about a pending review from our team and discuss any necessary improvements that need to be made to your change(s). 
    - If everything is successful, the team will merge your PR.

## Questions and Concerns

> If you have any questions or concerns about the guides or about Open Liberty, you can visit [Gitter for Open Liberty](https://gitter.im/OpenLiberty/) and post any inquires to any rooms. You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you may be facing.

Reviewer 3: ```suggestion
If you have any questions or concerns about the guides or about Open Liberty, you can visit [Gitter for Open Liberty](https://gitter.im/OpenLiberty/) and post your questions in the relevant rooms. You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you have.
```

# Contributing to Open Liberty Guides

Anyone can contribute to the Open Liberty guides and we welcome your contributions!

There are multiple ways to contribute: report bugs, fix bugs, contribute code, improve upon documentation, etc. 

## Raising Issues

> Please raise any bug reports on the [Open Liberty guide's repository's](https://github.com/OpenLiberty/guide-maven-intro/issues). For new topics, large updates to existing guides, or general suggestions and ideas, report your issue on the [Open Liberty common guides repository](https://github.com/OpenLiberty/guides-common/issues). Be sure to search the list to see if your issue has already been raised.

Reviewer: Instead of using `https://github.com/OpenLiberty/guide-maven-intro/issues`, try use `../../issues`, so that this file can be used for all guides without update this url

Reviewer: why cannot use `../.../issues`? 
I saw -> https://github.com/OpenLiberty/[guide-name]/issues, where [guide-name]
when I click on the link 404 error

Author: - Originally tried to make it in a way where reporting an issue on the repo is explicit.

Reviewer: will be addressed by gkwan-ibm-patch-1 branch

A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong. Provide as much context as possible so we can try to recreate the issue.

## Contributions 

> Contributing to the Open Liberty guides is simple. All you need to do is create your own branch to include your contribution(s). If you want to make updates to existing guides, create a new branch and open a pull request to the `qa` (quality assurance) branch. We first merge pull requests to the `qa` branch for staging before we merge to the `master` branch for publishing.

Reviewer: - [ ] "...the `qa` (quality assurance)... " -> "...the  quality assurance `qa`..."
- [ ] "We first merge... publishing." no need to mention

Author: In this case I believe the parenthesize are better left as "`qa` (quality assurance)".

Reviewer: let ID review

> After cloning the guide repository, create a new branch:

Reviewer: - [ ] I don't think this paragraph with the clone command is necessary

Author: You had mentioned to add similar instructions.  I removed it nonetheless.

```
git checkout -b [new-branch]
```

### Text or Content Contribution

> If you are contributing text of content changes or trivial code changes via a pull request, sign your work. The sign-off is a line at the end of the commit message that certifies that you wrote it or otherwise have the right to pass it on as an open source patch. The process is simple: if you can certify the [Developer Certificate of Origin (DCO)](https://developercertificate.org)),

Reviewer: - [ ] not mention "or trivial code changes.."
- [ ] suggest something like 
  - If you are contributing text or content changes and accept the DCO,  sign off your work by adding a sign-off statement in the commit message as following:
`Signed-off-by: John Doe <john.doe@example.com>`

then add a line to the end of the Git commit message:

```
Signed-off-by: Jane Williams <jane.williams@gmail.com>
```

Use your real name when you sign. We can't accept pseudonyms or anonymous contributions.

> Many Git UI tools have support for adding the `Signed-off-by` line to the end of your commit message. This line can be automatically added by the `git commit` command by using the `-s` option.

Author: This was just generally used across the other contribution like files under Open Liberty. I removed it nonetheless.

> Many Git UI tools have support for adding the `Signed-off-by` line to the end of your commit message. This line can be automatically added by the `git commit` command by using the `-s` option.

Reviewer: ok, you may add back  "Many Git UI .... -s option."

After we obtain the signed CLA, you are welcome to open a pull request, and the team will be notified for review. We ask you follow these steps through the submission process.

1. Open PR's against the `qa` branch, as we ensure changes pass our series of verification buckets before pushing to master.
2. A label will be added "CLA signed" or "CLA trivial" depending on the nature of the change.
3. A team of "reviewers" will be notified, will perform a review, and if approved will invoke a full integration build.

> 4. Based on the results of the build, and if further review is needed, more discussion will occur.

Reviewer: - [ ] If the reviewer agrees to the change, the PR will be merged to `qa` branch, otherwise the PR will be closed with an explanation and suggestion for followup.

Reviewer: - [ ] our process is different than OL, we cannot copy for OL contribution.md

> 5. If the reviewer is satisfied with the results, and agrees to the change, the PR will be merged to `qa` branch, otherwise the PR will be closed with an explaination and suggestion for followup.

Reviewer: - [ ] A team of "testers" will be tested your change. Based on the results, you will be required to fix your change or your change will be merge to the master branch for publishing.

Author: This looks like it contradicts: https://github.com/OpenLiberty/guide-maven-intro/pull/100#discussion_r470271723. Let me know if I misunderstood.

## Questions and Concerns

> If you have any questions or concerns about the guides, you can find Open Liberty on [Gitter](https://gitter.im/OpenLiberty/) and post any inquires about the guides to one of the rooms. You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you may be facing.

Reviewer: - [ ] suggest to 
  - If you have any questions or concerns about the guides or Open Liberty, you can visit [Gitter for Open Liberty](https://gitter.im/OpenLiberty/) and post any inquires to one of the rooms. You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you may be facing.

> # Contributing to Open Liberty Guides

Author: Addressed.

Anyone can contribute to the Open Liberty guides and we welcome your contributions!

There are multiple ways to contribute: report bugs, fix bugs, contribute code, improve upon documentation, etc. 

> ## Raising Issues

Author: Addressed.

> Please raise any bug reports on this [guide repository](../../issues). For new topics, large updates to existing guides, or general suggestions and ideas, report your issue on the [Open Liberty common guides repository](https://github.com/OpenLiberty/guides-common/issues). Be sure to search the list to see if your issue has already been raised.

Author: Addressed.

> A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong. Provide as much context as possible so we can try to recreate the issue.

Reviewer 4: >A good bug report is one that makes it easy for everyone to understand what you were trying to do and what went wrong.

To

>A good bug report makes it easy for everyone to understand what you were trying to do and what went wrong.

Author: Addressed.

## Contributions 

Contributing to an Open Liberty guide is simple. All you need to do is create your own branch to include your contribution and open a pull request to the `qa` (quality assurance) branch.

> ### Text or Content Contributions

Reviewer 4: >Text or Content Contributions

To

> Text or content contributions

Author: Addressed.

> If you are contributing text of content changes via a pull request, all you need is to certify the originality of the work via a [Developer Certificate of Origin (DCO)](https://developercertificate.org).

Reviewer 4: >If you are contributing text of content changes via a pull request, all you need is to certify the originality of the work via a [Developer Certificate of Origin (DCO)](https://developercertificate.org).

To

>If you are contributing text or content changes via a pull request, you need to certify that the originality of the work follows the [Developer Certificate of Origin (DCO)](https://developercertificate.org).

Author: Addressed.

Add a line to the end of the Git commit message to sign your work:

```
Signed-off-by: Jane Williams <jane.williams@gmail.com>
```

The sign-off is just a line at the end of the commit message that certifies that you wrote it or otherwise have the right to pass it on as an open source patch.

Use your real name when you sign. We can't accept pseudonyms or anonymous contributions. 

Many Git UI tools have support for adding the `Signed-off-by` line to the end of your commit message. This line can be automatically added by the `git commit` command by using the `-s` option.

If reviewers agree with your change, your change will be merged to the `master` branch for publishing.

> ### Code Contributions

Reviewer 4: >Code Contributions

To

>Code contributions

Author: Addressed.

> If you are contributing trivial code changes such as typos, redundant spaces, minor formatting and spelling errors, follow the same instructions under [Text or Content Contributions](./CONTRIBUTING.md#text-or-content-contributions).

Reviewer 4: >If you are contributing trivial code changes such as typos, redundant spaces, minor formatting and spelling errors, follow the same instructions under [Text or Content Contributions](./CONTRIBUTING.md#text-or-content-contributions).

To

>If you are contributing trivial code changes such as typos, redundant spaces, or minor formatting and spelling errors, follow the instructions under [Text or content contributions](./CONTRIBUTING.md#text-or-content-contributions).

Author: Addressed.

If you are contributing code changes via a pull request for non-trivial changes, you must sign off on the [Individual Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-individual.pdf). An example of a non-trivial change could be adding a new
microservice to the backend application of the guide. 

> If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf).

Reviewer 4: >If you are doing this as part of your job you may also wish to get your employer to sign a CCLA [Corporate Contributor License Agreement](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf).

To

>If you are contributing changes as part of your job, you may also wish to have your employer sign a [Corporate Contributor License Agreement (CCLA)](https://github.com/OpenLiberty/open-liberty/blob/master/cla/open-liberty-cla-corporate.pdf).

Author: Addressed.

> Instructions how to sign and submit these agreements are located at the top of each document. 

Reviewer 4: >Instructions how to sign and submit these agreements are located at the top of each document. 

To

>Instructions for how to sign and submit these agreements are located at the top of each document.

Author: Addressed.

> After we obtain the signed CLA, you are welcome to open a pull request against the `qa` branch, and the team will be notified for review. We ask you follow these steps through the submission process.

Reviewer 4: Add a colon after "submission process"

1. Open a pull request against the `qa` branch.
2. A "CLA signed" label will be manually added by the team.
3. The team will be notified and will review your change(s). 
    - If there are further changes to be made, the team will request changes on the pull request. 
    - If the team does not agree with the change, the PR will be closed with an explanation and suggestion for follow-up.
4. If the team approves, a full Open Liberty site build will be run.
5. Based on the results of the build:
    - If further review is needed, we will let you know about a pending review from our team and discuss any necessary improvements that need to be made to your change(s). 
    - If everything is successful, the team will merge your PR.

> ## Questions and Concerns

Reviewer 4: >Questions and Concerns

To

>Questions and concerns

Author: Addressed.

> If you have any questions or concerns about the guides or about Open Liberty, you can visit [Gitter for Open Liberty](https://gitter.im/OpenLiberty/) and post your questions in the relevant rooms. You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you have.

Reviewer 4: >You can also join the Open Liberty group on [Group.io](https://groups.io/g/openliberty) to discuss any issues you have.

To

>You can also join the Open Liberty group on [Groups.io](https://groups.io/g/openliberty) to discuss any issues you have.

Author: Addressed.
